emcc.py

```diff
@@ -1479,7 +1479,13 @@ def default_setting(name, new_default):
     if not shared.Settings.MINIMAL_RUNTIME:
       # In non-MINIMAL_RUNTIME, the core runtime depends on these functions to be present. (In MINIMAL_RUNTIME, they are
       # no longer always bundled in)
-      shared.Settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$demangle', '$demangleAll', '$jsStackTrace', '$stackTrace']
+      shared.Settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += [
+        '$keepRuntimeAlive',
```

Comment:
Tiny Unity currently uses the Fetch API and pthreads API with MINIMAL_RUNTIME, but it looks like in this PR, there will be a unDCEable core dependency added to this new `keepRuntimeAlive` abstraction when Fetch API or pthreads are used, but it is not available in MINIMAL_RUNTIME (since `shared.Settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE` is cleared below for MINIMAL_RUNTIME), and it increases code size for MINIMAL_RUNTIME even if not used.

(Most of the content in library.js and all of the content in library_browser.js are something that MINIMAL_RUNTIME never uses, so I am not worried about those parts adding unDCEable dependencies to this)

Reply:
I'm curious how Tiny Unity is able to use the fetch API in a pthread.  How does a thread keep itself alive long enough to receive the response back from the fetch request?   It seems that once a thread returns it will call `threadExit`, regardless of how many outstanding fetch requests there.  

Does it use `emscripten_unwind_to_js_event_loop` to keep the thread alive?    If so how does the thread eventually exit? 
 Does it call `pthread_exit` from the fetch callback?  Does that work?